test/api/groups.coffee

```diff
@@ -13,11 +13,70 @@ describe "Groups", ->
       registerNewUser ->
         User.findByIdAndUpdate user._id,
           $set:
-            "balance": 4
+            "balance": 10
           , (err, _user) ->
             done()
       , true
 
+    it "can create a public guild", (done) ->
+      request.post(baseURL + "/groups").send(
+        name: "TestGroup"
+        type: "guild",
+        privacy: "public"
+      ).end (res) ->
+        expectCode res, 200
+        guild = res.body
+        expect(guild.members.length).to.equal 1
+        expect(guild.leader).to.equal user._id
+        done()
+
+    it "can create a private guild", (done) ->
+      request.post(baseURL + "/groups").send(
+        name: "TestGroup"
+        type: "guild",
+        privacy: "private"
+      ).end (res) ->
+        expectCode res, 200
+        guild = res.body
+        expect(guild.members.length).to.equal 1
+        expect(guild.leader).to.equal user._id
+        done()
+
+    it "prevents user from creating a guild when the user has 0 gems", (done) ->
+      registerNewUser (err, user) ->
```

Comment:
In general, I'd put setup like this in a before block. To do that, you'll probably need an extra describe or context block (under the hood, context is just an alias for describe).

For this case, I think it'd make sense to use a context block and it'd be written something like:

``` javascript
context "creating a guild when the user has 0 gems", ->
  before (done) ->
    registerNewUser done, true

  it "does not create", (done) ->
    request.post(baseURL + "/groups").send(
        name: "TestGroup"
        type: "guild",
    )
    .end (res) ->
      expectCode res, 401
      expect(res.body.err).to.equal 'Not enough gems!'
      done()